packages/babel-plugin-transform-modules-umd/src/index.js

```diff
@@ -27,7 +27,7 @@ const buildWrapper = template(`
 
       GLOBAL_TO_ASSIGN;
     }
-  })(this, function(IMPORT_NAMES) {
+  })(globalThis || window || this, function(IMPORT_NAMES) {
```

Comment:
Wouldn't this throw a `globalThis is not defined` error in browsers with modules supprot (where the code is executed in strict mode) but without `globalThis` support?

Those browsers are Edge 16, Firefox 60-64, Chrome 61-70, Safari 10.1-12.0 and Opera 48-57

Reply:
```suggestion
  })(
    typeof globalThis === "object" ? globalThis
      : typeof window === "object" ? window
      : this,
    function(IMPORT_NAMES) {
```

Reply:
Definitely need Nicolo's suggestion, but we should use `self` instead of `window` to support workers as well.

Reply:
````javascript
typeof globalThis === "object" ? globalThis
: typeof self === "object" ? self
: typeof window === "object" ? window
: typeof global === "object" ? global
: this
```
Is that too overboard on trying?  I think there are environments where `window` is defined but not `self` or `globalThis` (older browsers), and older versions of NodeJS only define `global` I believe.  Does it make sense to fallback to `this` at the end?  Perhaps for test environments the `this` fallback is valuable?

Reply:
I think just `self` is enough. IE has supported it forever. As for `global`, let's assume people can just load a `globalThis` polyfill in node. Client side we have to worry about bytes.

Reply:
The suggested changes have been applied and the code rebased.  Anyone can feel free to takeover and finish this up (e.g., update the tests).  I'm on Windows and this repo appears to have a number of issues with Windows so I doubt I'll be able to finish it.